cli/command/container/run.go

```diff
@@ -259,10 +259,11 @@ func runRun(dockerCli *command.DockerCli, flags *pflag.FlagSet, opts *runOptions
 // reportError is a utility method that prints a user-friendly message
 // containing the error that occurred during parsing and a suggestion to get help
 func reportError(stderr io.Writer, name string, str string, withHelp bool) {
+	str = strings.TrimSuffix(str, ".") + "."
 	if withHelp {
-		str += ".\nSee '" + os.Args[0] + " " + name + " --help'"
+		str += "\nSee '" + os.Args[0] + " " + name + " --help'"
 	}
-	fmt.Fprintf(stderr, "%s: %s.\n", os.Args[0], str)
+	fmt.Fprintf(stderr, "%s: %s\n", os.Args[0], str)
```

Comment:
Since the error returned may also include a period, so there is some cases that message shows two duplicated periods, like `docker: Error response from daemon: Conflict. The container name "/allen7" is already in use by container 7fcfde24ef171a7e690366b8376d1d2facca1d755dabbe357e80b70dec138b78. You have to remove (or rename) that container to be able to reuse that name..` with two periods at the end.
As a result, I try to use `str = strings.TrimSuffix(str, ".") + "."` to make sure that there is always only one period.